### DISCUSSION TITLE
Strage behavior with casts

### DISCUSSION THREAD

**Discussion by etdte** [laravel/framework#36290] [Q&A]
- Laravel Version: 8.28.0
- PHP Version: 8.0.2
- Database Driver & Version: sqlite version 3.32.3. Postgres 13.0

### Description:
I've noticed strage behavior of casts, when same cast class works perfectly when `$guarded` array of the model is emply, and does not work when saving model which has values in `$guarded` attribute. 

### Steps To Reproduce:
 1. Clone https://github.com/etdte/poc-cast-issue
 2. Install Composer dependecies. 
 3. Run ProofTest test `./vendor/bin/phpunit --filter=ProofTest`.

**Comment by driesvints** [comment]
Please re-create the repo with any custom changes committed separately from the base Laravel skeleton.

**Comment by etdte** [comment]
Here is a new one, with every change commited separetely.

https://github.com/etdte/laravel-issue-36285
